File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<string> babbling) {
    int answer = 0;
    
    for (int i = 0; i < babbling.size(); i++) {
        string cur = babbling[i];
        int prevIdx = -1;
        bool flag = true;
        
        for (int j = 0; j < cur.size(); j++) {
            if (cur.substr(j, 3) == "aya" && prevIdx != 1) {
                prevIdx = 1;
                j += 2;
            } else if (cur.substr(j, 2) == "ye" && prevIdx != 2) {
                prevIdx = 2;
                j += 1;
            } else if (cur.substr(j, 3) == "woo" && prevIdx != 3) {
                prevIdx = 3;
                j += 2;
            } else if (cur.substr(j, 2) == "ma" && prevIdx != 4) {
                prevIdx = 4;
                j += 1;
            } else {
                flag = false;
                break;
            }
        }
        
        if (flag) {
            answer += 1;
        }
    }
    
    return answer;
}

File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.py
def solution(babbling):
    answer = 0

    for word in babbling: 
        flag = True;
        prevIdx = -1
        while word:
            if (word[:3] == "aya" and prevIdx != 1):
                word = word[3:]
                prevIdx = 1
            elif (word[:2] == "ye" and prevIdx != 2):
                word = word[2:]
                prevIdx = 2
            elif (word[:3] == "woo" and prevIdx != 3):
                word = word[3:]
                prevIdx = 3
            elif (word[:2] == "ma" and prevIdx != 4):
                word = word[2:]
                prevIdx = 4
            else:
                flag = False
                break
            print(word)
        
        if (flag):
            answer += 1
        prevIdx = -1  
                
        
    return answer